Use the new TIFFFieldSetGetSize API from libtiff 4.4 instead of the
private and removed _TIFFDataSize declared in the unbundling patch.

Patch taken from Fedora:

  https://src.fedoraproject.org/rpms/freeimage/blob/rawhide/f/freeimage-libtiff44.patch

diff -rupN --no-dereference freeimage-svn-r1889-FreeImage-trunk/Source/Metadata/XTIFF.cpp freeimage-svn-r1889-FreeImage-trunk-new/Source/Metadata/XTIFF.cpp
--- freeimage-svn-r1889-FreeImage-trunk/Source/Metadata/XTIFF.cpp	2022-06-23 11:56:32.561043826 +0200
+++ freeimage-svn-r1889-FreeImage-trunk-new/Source/Metadata/XTIFF.cpp	2022-06-23 11:56:32.764043827 +0200
@@ -747,7 +747,7 @@ tiff_write_exif_tags(TIFF *tif, TagLib::
 				continue;
 			}
 			// type of storage may differ (e.g. rationnal array vs float array type)
-			if((unsigned)_TIFFDataSize(tif_tag_type) != FreeImage_TagDataWidth(tag_type)) {
+			if((unsigned)TIFFFieldSetGetSize(fld) != FreeImage_TagDataWidth(tag_type)) {
 				// skip tag or _TIFFmemcpy will fail
 				continue;
 			}
